### PR TITLE
fix: k3s extra package download

### PR DESCRIPTION
**Pull request by DanStough**

### CODE REVIEW

**Review by emosbaugh — COMMENTED** [review]
this all seems very brittle

File: scripts/distro/rke2/distro.sh [hunk 1]
```diff
@@ -60,6 +60,7 @@ function rke2_reset() {
 
 function rke2_containerd_restart() {
     rke2_restart
+    sleep 10        # In practice it takes some time for the workloads to end.
```
Comment (emosbaugh) [inline scripts/distro/rke2/distro.sh:63]: would it be possible to monitor the systemd service?

Comment (DanStough) [reply scripts/distro/rke2/distro.sh]: Yes, pushed an update.

File: scripts/common/common.sh [hunk 2]
```diff
@@ -626,3 +626,38 @@ function get_kurl_install_directory_flag() {
...
+function restart_systemd_and_wait() {
+    local serviceName=$1
+    
+    local pid="$(systemctl show --property MainPID $serviceName | cut -d = -f2)"
+
+    echo "Restarting $serviceName..."
+    systemctl restart $serviceName
+
+    if ! spinner_until 120 systemd_restart_succeeded $pid $serviceName; then
+        bail "Could not successfully restart systemd service $serviceName"
```
Comment (emosbaugh) [inline scripts/common/common.sh:659]: `journalctl -xe` output here could be helpful

File: scripts/common/common.sh [hunk 3]
```diff
@@ -626,3 +626,38 @@ function get_kurl_install_directory_flag() {
...
+function systemd_restart_succeeded() {
+    local oldPid=$1
+    local serviceName=$2
+
+    if ! systemctl is-active --quiet $serviceName; then
+        return 1
+    fi
+
+    local newPid="$(systemctl show --property MainPID $serviceName | cut -d = -f2)"
+    if [ "$newPid" == "$oldPid" ]; then
```
[the remark this answers is not in the document]
Comment (DanStough) [reply scripts/common/common.sh:639]: This is probably just a bug. Will update.

[the remark this answers is not in the document]
Comment (DanStough) [reply scripts/common/common.sh]: Also looked this up:
- Double equals is pattern matching
- Single equals is equivalent check

**Pull request merged**